File: Backend/src/main/java/com/kas/online_book_shop/service/BookService.java
package com.kas.online_book_shop.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.kas.online_book_shop.enums.BookState;
import com.kas.online_book_shop.model.Book;
import com.kas.online_book_shop.model.BookCategory;
import com.kas.online_book_shop.model.BookCollection;

public interface BookService {

    Book getBookById(Long id);

    Page<Book> getAllBooks(Pageable pageable);

    List<Book> getAllBooks();

    Book saveBook(Book book);

    Book updateBook(Book book);

    void deleteBook(Long id);

    void changeBookState(Long id);

    Page<Book> getBookByCategoriesAndPriceRange(List<BookCategory> categories, int min, int max, Pageable pageable);

    Page<Book> getBooksByCollectionAndPriceRanges(BookCollection collection,int min, int max,Pageable pageable);

    Page<Book> getBooksByName(String name, Pageable pageable);

    void addBookToCollection(Long bookId, Long collectionId);

    Page<Book> queryBook(String title, BookState state, BookCollection collection, Pageable pageable);
}

File: Frontend/re-admin/src/service/BookService.js
import axios from 'axios'

const BOOK_BASE_URL = "http://localhost:8081/api/v1/book";

const getAllBooks = () => {
    return axios.get(BOOK_BASE_URL + '/get-all');
}

const getBookById = (bookId) => {
    return axios.get(BOOK_BASE_URL + '/' + bookId);
}

const updateBook = (data) => {
    return axios.put(BOOK_BASE_URL, data);
}

const addBook = (data) => {
    return axios.post(BOOK_BASE_URL, data);
}

const deleteBook = (bookId) => {
    return axios.delete(BOOK_BASE_URL + '/' + bookId);
}

export {getAllBooks, getBookById, updateBook, addBook, deleteBook}